src/site/pages/digital/src/containers/ExprToCircuitPopup/index.tsx

```diff
@@ -0,0 +1,182 @@
+import {useState} from "react";
+import {connect}  from "react-redux";
+
+import {CreateAddGroupAction} from "core/actions/addition/AddGroupActionFactory";
+
+import {Popup}   from "shared/components/Popup";
+
+import {SharedAppState}    from "shared/state";
+import {CloseHeaderPopups} from "shared/state/Header";
+import {HeaderPopups}      from "shared/state/Header";
+
+import {OrganizeMinDepth} from "core/utils/ComponentOrganizers";
+
+import {GroupAction}             from "core/actions/GroupAction";
+import {CreateDeselectAllAction,
+        CreateGroupSelectAction,
+        SelectAction,}            from "core/actions/selection/SelectAction";
+import {PlaceAction}             from "core/actions/addition/PlaceAction";
+import {CreateICDataAction}      from "digital/actions/CreateICDataAction";
+
+import {DigitalCircuitInfo}     from "digital/utils/DigitalCircuitInfo";
+import {DigitalComponent}       from "digital/models/DigitalComponent";
+import {ICData}                 from "digital/models/ioobjects/other/ICData";
+import {IC}                     from "digital/models/ioobjects/other/IC";
+import {LED}                    from "digital/models/ioobjects/outputs/LED";
+import {ConstantLow}            from "digital/models/ioobjects/inputs/ConstantLow";
+import {ConstantHigh}           from "digital/models/ioobjects/inputs/ConstantHigh";
+import {Button}                 from "digital/models/ioobjects/inputs/Button";
+import {Switch}                 from "digital/models/ioobjects/inputs/Switch";
+import {Clock}                  from "digital/models/ioobjects/inputs/Clock";
+import {GenerateTokens,
+        ExpressionToCircuit}    from "digital/utils/ExpressionParser";
+import {FormatMap}    from "digital/utils/ExpressionParserConstants";
+
+import "./index.scss";
+
+
+type OwnProps = {
+    info: DigitalCircuitInfo;
+}
+type StateProps = {
+    curPopup: HeaderPopups;
+}
+type DispatchProps = {
+    CloseHeaderPopups: typeof CloseHeaderPopups;
+}
```

Comment:
We recently refactored how to pass props/actions from Redux, take a look at the other containers and it should be fairly straightforward and make the code much nicer!